OpenRA.Game/Map/CellLayer.cs

```diff
@@ -34,15 +35,35 @@ public CellLayer(TileShape shape, Size size)
 			entries = new T[size.Width * size.Height];
 		}
 
-		public void CopyValuesFrom(CellLayer<T> anotherLayer)
+		// provides a default instance for this type of Layer
+		public static void Init(CellLayer<T> defaultLayer)
 		{
-			if (Size != anotherLayer.Size || Shape != anotherLayer.Shape)
+			defaultCellInfoLayer = defaultLayer;
+		}
+
+        public static void Init(Func<int, int, T> creator, int mapWidth, int mapHeight, TileShape tileShape)
```

Comment:
You got a point on the "reusability", but the problem is performance. Creating 128*128 news everytime you need to pathfind is costly, and cloning(arraycopy) is way faster.

So yeah, we need a prototype variable to clone, and yeah, also a way (maybe an Id and a dictionary?) To handle the retrieval and storage of exclusive prototypes. I'll think about that.